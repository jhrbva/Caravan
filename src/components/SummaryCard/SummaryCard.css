.flex-container {
	padding: 0;
	margin: 0;
	list-style: none;
	-ms-box-orient: horizontal;
	display: -webkit-box;
	display: -moz-box;
	display: -ms-flexbox;
	display: -moz-flex;
	display: -webkit-flex;
	display: flex;
}

.wrap {
	-webkit-flex-wrap: wrap;
	flex-wrap: wrap;
}

.flex-item {
	padding: 5px;
	margin: 10px;
}
.card {
	box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
	margin-bottom: 30px;
	margin-right: 0px;
}
.card:hover {
	box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.2);
}

.cardbody {
	text-align: left;
	overflow: scroll;
	overflow-wrap: break-word;
	word-wrap: break-word;
	hyphens: auto;
}
.cardtitle {
	margin-left: 40px;
}
.material-icons {
	font-size: 20px;
	position: absolute;
	margin-top: -10px;
	color: rgb(53, 130, 202);
}
