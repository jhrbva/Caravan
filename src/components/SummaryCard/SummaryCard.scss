@import '../../utilities/colorPalette.scss';

.flex-container {
	padding: 0;
	margin: 0;
	list-style: none;
	-ms-box-orient: horizontal;
	display: -webkit-box;
	display: -moz-box;
	display: -ms-flexbox;
	display: -moz-flex;
	display: -webkit-flex;
	display: flex;
}

.wrap {
	-webkit-flex-wrap: wrap;
	flex-wrap: wrap;
}

.flex-item {
	padding: 5px;
	margin: 10px;
}

.card.card {
	width: 18rem;
	height: 14rem;

	margin: 15px auto;
	border-radius: 30px;

	box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
}

.card.card:hover {
	box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.2);
}

.card-title.h5 {
	text-align: left;
	font-size: 21px;
	margin: 15px 0 15px 60px;
}

.card-text {
	font-size: 18px;
	font-weight: 500;
	color: white;
}

.card-body {
	overflow: scroll;
	background-color: $dream-blue-fade;
	border-bottom-left-radius: 30px;
	border-bottom-right-radius: 30px;
	display: flex;
	justify-content: center;
	align-items: center;
}

.card .material-icons {
	position: absolute;
	margin: 5px 10px;
	color: $dream-blue-fade;
}
.MuiSvgIcon-fontSizeLarge {
	font-size: 2rem !important;
}
