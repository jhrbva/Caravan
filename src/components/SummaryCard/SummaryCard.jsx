import React, { useState } from 'react';
import { Card, Modal } from 'react-bootstrap';

import './SummaryCard.scss';
import TripDetails from '../TripDetails/TripDetails';

const SummaryCard = (props) => {
	const [show, setShow] = useState(false);
	const handleClose = () => setShow(false);
	const handleShow = () => setShow(true);
	const { trip, host, members, reststops, icon, isYourTrips, rerender } = props;
	console.log(props);
	const text = isYourTrips
		? `You are going to ${trip.destination}.`
		: `You are invited to ${trip.destination} by ${trip.username}.`;

	return (
		<>
			<Card onClick={handleShow}>
				<i className='material-icons'>{icon}</i>
				<Card.Title>{trip.trip_title}</Card.Title>
				<Card.Body>
					<Card.Text>{text}</Card.Text>
				</Card.Body>
			</Card>
			<Modal show={show} onHide={handleClose}>
				<TripDetails
					userid={props.userid}
					isYourTrips={isYourTrips}
					trip={trip}
					host={host}
					members={members}
					reststops={reststops}
					rerender={rerender}
				/>
			</Modal>
		</>
	);
};

export default SummaryCard;
