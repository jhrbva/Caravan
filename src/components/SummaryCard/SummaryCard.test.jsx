import React from 'react';
import { shallow } from 'enzyme';
// import { renderHook, act } from '@testing-library/react-hooks';

import SummaryCard from './SummaryCard';

const trip = {
	trip_title: 'Wonders of the Canyon',
	tripdate: '2020-06-01T13:55:06.000Z',
	trip_description: 'Join in as we travel to the Grand Canyon',
	startlocation: '123 OneTreeHill St.',
	destination: 'Grand Canyon, Grand Canyon Ave.',
};

const host = {
	username: 'conniecode',
};
const members = [{
	username: 'jcode'
}];

const reststops = [{ location: 'Wawa' }, { location: 'Loves' }];

describe('SummaryCard', () => {
	it('should match the snapshot', () => {
		const wrapper = shallow(<SummaryCard trip={trip} host={host} members={members} reststops={reststops}/>);
		expect(wrapper).toMatchSnapshot();
	});
});

// TODO: Test React Hooks for handleShow & handleClose using react-hook testing library.
