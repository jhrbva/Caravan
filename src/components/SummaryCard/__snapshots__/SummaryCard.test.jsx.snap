// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryCard should match the snapshot 1`] = `
<Fragment>
  <Card
    body={false}
    onClick={[Function]}
    style={
      Object {
        "width": "18rem",
      }
    }
  >
    <CardBody>
      <CardTitle>
        Wonders of the Canyon
      </CardTitle>
      <CardText>
        Grand Canyon, Grand Canyon Ave.
      </CardText>
    </CardBody>
  </Card>
  <Bootstrap(Modal)
    onHide={[Function]}
    show={false}
  >
    <TripDetails
      trip={
        Object {
          "destination": "Grand Canyon, Grand Canyon Ave.",
          "startlocation": "123 OneTreeHill St.",
          "trip_description": "Join in as we travel to the Grand Canyon",
          "trip_title": "Wonders of the Canyon",
          "tripdate": "2020-06-01T13:55:06.000Z",
        }
      }
    />
  </Bootstrap(Modal)>
</Fragment>
`;
